api/proto/auth/delegator.go

```diff
@@ -0,0 +1,28 @@
+package auth
+
+import (
+	"context"
+
+	"github.com/gravitational/teleport/lib/events"
+)
+
+type contextKey string
+
+const ContextDelegator contextKey = events.AccessRequestDelegator
+
+// WithDelegator creates a child context with the AccessRequestDelegator
+// value set.  Optionally used by AuthServer.SetAccessRequestState to log
+// a delegating identity.
+func WithDelegator(ctx context.Context, delegator string) context.Context {
+	return context.WithValue(ctx, ContextDelegator, delegator)
+}
+
+// getDelegator attempts to load the context value AccessRequestDelegator,
+// returning the empty string if no value was found.
+func getDelegator(ctx context.Context) string {
```

Comment:
Maybe we can move this into `services` or somewhere else, that would be one less place where we have to import `events`.

Reply:
But then we would need to import `services`. I'll probably do a 3rd PR to handle imports of Events, Wrappers, and possibly the other proto packages.

Reply:
@fspmarshall Can we move `events.AccessRequestDelegator` into a constant? And maybe then `events.AccessRequestDelegator` just picks up the value from the constants?

We're trying to break the dependency on the `events` package here.

Reply:
Yeah, no problem.  In fact, the context key can be anything so long as `getDelegator` and `WithDelegator` agree on which key to use.  If anything, its probably preferable to use a [typed key](https://medium.com/golangspec/globally-unique-key-for-context-value-in-golang-62026854b48f) rather than a simple string constant.